tools/benchmark/src/main/java/com/datadog/benchmark/internal/MetricHttpClient.kt

```diff
@@ -0,0 +1,163 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.benchmark.internal
+
+import android.util.Log
+import com.datadog.benchmark.DatadogExporterConfiguration
+import com.datadog.benchmark.internal.model.MetricContext
+import com.google.gson.JsonElement
+import io.opentelemetry.sdk.metrics.data.MetricData
+import okhttp3.CacheControl
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.ResponseBody
+import java.io.IOException
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+internal class MetricHttpClient(
+    private val context: MetricContext,
+    private val exporterConfiguration: DatadogExporterConfiguration
+) {
+
+    private val okHttpClient: OkHttpClient = OkHttpClient.Builder()
+        .callTimeout(TIMEOUT_SEC, TimeUnit.SECONDS)
+        .connectTimeout(TIMEOUT_SEC, TimeUnit.SECONDS)
+        .readTimeout(TIMEOUT_SEC, TimeUnit.SECONDS)
+        .writeTimeout(TIMEOUT_SEC, TimeUnit.SECONDS)
+        .cache(null)
+        .build()
+
+    private val requestBodyBuilder = RequestBodyBuilder(context)
+
+    fun uploadMetric(metrics: List<MetricData>) {
+        val response = postRequest(
+            exporterConfiguration = exporterConfiguration,
+            requestBodyBuilder.buildJsonElement(metrics)
+        )
+        if (response.isSuccess) {
+            Log.i(BENCHMARK_LOGCAT_TAG, "${metrics.size} Metrics data uploaded.")
+        } else {
+            Log.e(BENCHMARK_LOGCAT_TAG, "Metrics data failed to upload. ${response.exceptionOrNull()}")
+        }
+    }
+
+    private fun postRequest(
```

Comment:
In my opinion `postRequest` will also work with traces because it takes generic parameters which are `DatadogExporterConfiguration` and `JsonElement`.

Maybe the name of the class is misleading, it shouldn't be dedicated to metrics, so I rename if `DatadogHttpClient` if it seems better to you